lib/sierra-models/bib.js

```diff
@@ -51,23 +55,51 @@ class SierraBib extends SierraBase {
     return { suppressed: Boolean(rationale), rationale }
   }
 
-  isResearch () {
-    let locationCodes = []
-    const locations = this.locations
-    if (locations && Array.isArray(locations)) {
-      locationCodes = locations.map(location => location.code)
+  getIsResearchWithRationale () {
+    // Partner bibs are Research
+    if (this.nyplSource && this.nyplSource !== 'sierra-nypl') {
+      return { isResearch: true, rationale: 'Is partner bib' }
     }
 
+    // Gather location codes on the bib:
+    const locationCodes = this.locations && Array.isArray(this.locations)
+      ? this.locations.map((location) => location.code)
+      : []
+
     if (locationCodes.includes('none') || locationCodes.includes('os')) {
-      return true
+      return { isResearch: true, rationale: 'Locations include none/os' }
+    } else {
+      // Determine locations that are exclusively 'Branch'
+      const branchLocations = locationCodes
+        .filter((code) => locationHasExclusiveCollectionType(code, 'Branch'))
+      // If bib has any locations with just "Branch" collection type, bib is not research:
+      if (branchLocations.length > 0) {
+        return {
+          isResearch: false,
+          rationale: `Has branch locations: ${branchLocations.join(', ')}`
```

Comment:
are there any bibs with some research and some circ items?

Reply:
No. Or if there are, we don't support then.